src/softmax.jl

```diff
@@ -55,33 +73,33 @@ function softmax!(out::AbstractArray{T}, x::AbstractArray; dims = 1) where {T}
     else
         @. out = ifelse(isequal(max_,Inf), ifelse(isequal(x,Inf), 1, 0), exp(x - max_))
     end
-    out ./= sum(out; dims = dims)  # could re-use max_ when dims != (:) and eltype(x) == T.
+    y = out ./= sum(out; dims = dims)
```

Comment:
```suggestion
    out ./= sum(out; dims = dims)
    return out
```

Reply:
My thinking here, and with `dx = if` just below, is essentially to comment that we are calling the input x and the output y. Maybe it's too weird?

Reply:
removed